Source/WebGPU/WGSL/AST/ASTTypeDecl.h

```diff
@@ -74,9 +81,16 @@ class NamedType final : public TypeDecl {
 
     Kind kind() const override;
     const String& name() const { return m_name; }
+    TypeDecl* maybeResolvedReference() const { return m_resolvedReference.get(); }
 
 private:
+    void resolveTypeReference(Ref<TypeDecl>&& typeDecl)
+    {
+        m_resolvedReference = WTFMove(typeDecl);
+    }
```

Comment:
I must not be understanding this. The `NamedType` is a superclass of a struct declaration, right? I'm confused about why a TypeDecl has a ref to another TypeDecl for the resolved reference... Why isn't this specific to typedefs? Also, how do type parameters work?

Reply:
`NamedType` isn't a superclass of `StructDecl`. This is to cover the following:

```
struct S { }
fn f(s: S) { }
```

What happens now is:
1. We read the parameter `s` from `f`,  then get its type `S`;
2. We want to be able to access the actual struct so we can read its fields' types.

Before this patch, all we got was a `NamedType { name = 'S' }`, now we have a way to go from that to `StructType { .members = { ... } }` so we can read the actual fields of `S`.

This is unrelated to type parameters, that will come later.